tech-papers/lbft-verification/lbft-verification.tex

```diff
@@ -0,0 +1,199 @@
+
+\subsection{Waypoint structure}
+\label{waypoint-structure}
+In order to provide a client with the initial trusted root hash and validators, it is enough for the waypoint to include the information about a LedgerInfo on the epoch boundary (the one that must include the validator set of the next epoch).
+
+The waypoint representation remains a short string that can be easily included in a config or copied from an association website.
+
+\begin{algorithm}[H]
+\myalgorithm
+{
+\Iblock{\rust{Waypoint}}{
+	   \tcp{The version of the reconfiguration transaction that is being approved by this waypoint.}
+        \rust{version: u64}\;
+        \tcp{The hash of the chosen fields of LedgerInfo (including the next validator set).}
+        \rust{value: HashValue}
+}
+}
+\end{algorithm}
+
+
+The fields of a LedgerInfo that are included in the waypoint value are \texttt{epoch}, \texttt{root\_hash}, \texttt{version}, \texttt{timestamp}, and \texttt{next\_validator\_set}. Note that we cannot include all the fields because they might not be deterministic: different clients might observe different values of e.g., \texttt{consensus\_block\_id} of the latest \texttt{LedgerInfo} in an epoch.
+
+The textual representation of a Waypoint is  "version:hex encoding of the root hash". For example, the current testnet genesis waypoint is \newline \texttt{"0:a17035961b97359bcc22b628c1e269f543b265220d22a17e4a5b4e3aed86855b"}.
+
+\subsection{Verifying the latest LedgerInfo with waypoints}
+\label{waypoint-li-verification}
+As described in Section~\ref{verification-epochs}, an \texttt{UpdateToLatestLedgerResponse} message carries a data structure for verifying validator changes, \texttt{ValidatorChangeProof}, which contains a list of all the epoch change \texttt{LedgerInfos}. A client verifies this list in an iterative fashion starting with its initial trusted \texttt{ValidatorSet}.
+
+\begin{figure}[ht]
+	\centering
+	\includegraphics[width=\textwidth]{figures/waypoint-verification.pdf}
+	\caption{\footnotesize{The first LedgerInfo in the \texttt{ValidatorChangeProof} is verified via a local waypoint.}}
+	\label{fig:waypoint-verification}
+\end{figure}
+
+Waypoint verification is working slightly differently as shown in Figure~\ref{fig:waypoint-verification}.
+A waypoint version corresponds to the last version of an epoch (version $0$ being somewhat special and corresponding to the last and only version of epoch $0$).
+Hence, the \texttt{ValidatorChangeProof} in the response starts with the end-of-epoch \texttt{LedgerInfo} corresponding to the waypoint: this very first \texttt{LedgerInfo} is verified using the waypoint. The rest of the epoch changes is verified in a regular manner.
+
+\subsection{State synchronization with waypoints}
+\label{waypoint-state-sync}
+Unfortunately, the verification of the latest \texttt{LedgerInfo} by itself might not be enough for secure state synchronization.
```

Comment:
In the example below, why can't B provide C with the same proof for W_80 that B obtained?